crates/db/src/lib.rs

```diff
@@ -61,3 +64,131 @@ impl From<&str> for DatabaseFlavour {
         }
     }
 }
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct StringHolder(pub String);
+
+impl AsByteBuffer for StringHolder {
+    fn size_bytes(&self) -> usize {
+        self.0.len()
+    }
+
+    fn with_byte_buffer<R, F: FnMut(&[u8]) -> R>(&self, mut f: F) -> Result<R, EncodingError> {
+        Ok(f(self.0.as_bytes()))
+    }
+
+    fn make_copy_as_vec(&self) -> Result<Vec<u8>, EncodingError> {
+        Ok(self.0.as_bytes().to_vec())
+    }
+
+    fn from_sliceref(bytes: SliceRef) -> Result<Self, DecodingError> {
+        Ok(Self(
+            String::from_utf8(bytes.as_slice().to_vec()).expect("Invalid UTF-8"),
```

Comment:
Huh, weird to see an `.expect("Invalid UTF-8")` in a function that's `Result<..., DecodingError>` :D I know this `DecodingError` is a custom error type, but even so.

Reply:
I have two thoughts

a) DecodingError could have the data integrity thing but

b) Bad UTF8 there is... no, that's no good. And we should panic. It's not an error we're going to recover nice from. And I want to know in the backtrace exactly where it happened.

I'll leave it for now but we can revisit later maybe